micrometer-spring-legacy/src/main/java/io/micrometer/spring/advice/TimedMethodInterceptor.java

```diff
@@ -0,0 +1,88 @@
+package io.micrometer.spring.advice;
+
+import io.micrometer.core.annotation.Timed;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.Timer.Builder;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.springframework.core.annotation.AnnotatedElementUtils;
+import org.springframework.util.ReflectionUtils;
+
+import java.lang.reflect.Method;
+
+public class TimedMethodInterceptor implements MethodInterceptor {
+    public static final String DEFAULT_METRIC_NAME = "method_timed";
```

Comment:
should this possibly be shared with the aspectJ based metric name?

Reply:
I was thinking about replacing `TimedAspect` with this implementation, but if not I'll change it.